app/models/membership_search.rb

```diff
@@ -1,51 +1,70 @@
 # frozen_string_literal: true
 
 # MembershipSearch provides the `search`` extension method
-# for Roles.membership_dataset to allow text searching
+# for Role membership datasets to allow text searching
 # and paging of Role members
-module MembershipSearch
-  # @param search [String] - a search term in the resource id
-  def search(search: nil, kind: nil)
-    filter_kind(kind).textsearch(search)
-  end
-
-  def filter_kind(kind)
-    return self unless kind
+class MembershipSearch < Module
```

Comment:
`class` already inherits from `Module`, so it's unnecessary (and kind of confusing) to say it again.